processing_options.go

```diff
@@ -952,6 +966,10 @@ func parsePath(ctx context.Context, r *http.Request) (context.Context, error) {
 		imageURL, po, err = parsePathAdvanced(parts[1:], headers)
 	}
 
+	if !isAllowedSource(imageURL) {
+		return ctx, newError(404, fmt.Sprintf("Invalid source"), msgInvalidSource)
+	}
+
```

Comment:
This should be placed after the following error check

Reply:
Removed